src/win/path_util.cc

```diff
@@ -4,25 +4,28 @@
  * Copyright (c) 2018, Microsoft Corporation (MIT License).
  */
 
-#include <nan.h>
 #include <Shlwapi.h> // PathCombine
 
 #include "path_util.h"
 
 namespace path_util {
 
-const wchar_t* to_wstring(const Nan::Utf8String& str) {
-  const char *bytes = *str;
-  int sizeOfStr = MultiByteToWideChar(CP_UTF8, 0, bytes, -1, NULL, 0);
-  if (sizeOfStr <= 0) {
-    return L"";
+std::wstring to_wstring(const Napi::String& str) {
+  const std::u16string & u16 = str.Utf16Value();
+  return std::wstring(u16.begin(), u16.end());
+}
+
+std::string wstring_to_string(const std::wstring &wide_string) {
+  if (wide_string.empty()) {
+    return "";
   }
-  wchar_t *output = new wchar_t[sizeOfStr];
-  int status = MultiByteToWideChar(CP_UTF8, 0, bytes, -1, output, sizeOfStr);
-  if (status == 0) {
-    return L"";
+  const auto size_needed = WideCharToMultiByte(CP_UTF8, 0, &wide_string.at(0), (int)wide_string.size(), nullptr, 0, nullptr, nullptr);
```

Comment:
prefer c++ cast over c-style cast

Reply:
According to my testing, it's the best way to process DIFFERENT CODE PAGE correctly, there are some C++ style solution like `std::string str(wide.begin(),wide.end())` in [stackoverflow.com/q/4804298](https://stackoverflow.com/q/4804298/) however them doesn't care about old Windows with CJK code page.

This function is only used under Windows so IMO it's ok.